### ISSUE TITLE
DB connection crashes

### ISSUE THREAD

**Issue by gerbyzation** [okfn/docker-ckan#2]
When going to `/dataset` or performing a search the CKAN service crashes with ` LOG:  unexpected EOF on client connection with an open transaction`.

```
ckan          | [prerun] Sysadmin user exists, skipping creation
ckan          | Starting subprocess with file monitor
solr          | 1930189 INFO  (qtp1989972246-17) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&rows=1&wt=json} hits=0 status=0 QTime=0
solr          | 1930197 INFO  (qtp1989972246-21) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/admin/file params={file=schema.xml} status=0 QTime=0
solr          | 1930548 INFO  (qtp1989972246-19) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={q=*:*&rows=1&wt=json} hits=0 status=0 QTime=0
solr          | 1930554 INFO  (qtp1989972246-18) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/admin/file params={file=schema.xml} status=0 QTime=0
ckan          | Starting server in PID 27.
ckan          | 2018-01-22 15:04:49,113 INFO  [ckan.lib.base]  /api/3/action/status_show render time 0.018 seconds
ckan          | 2018-01-22 15:04:59,189 INFO  [ckan.lib.base]  /api/3/action/status_show render time 0.005 seconds
solr          | 1942913 INFO  (qtp1989972246-12) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={facet.limit=50&q=*:*&facet.field=organization&facet.field=groups&facet.field=tags&facet.field=res_format&facet.field=license_id&fl=id+validated_data_dict&start=0&sort=views_recent+desc&fq=%2Bcapacity:public+capacity:"public"&fq=%2Bsite_id:"default"&fq=%2Bstate:active&fq=%2Bpermission_labels:("public")&facet.mincount=1&rows=5&facet=true&wt=json} hits=0 status=0 QTime=1
ckan          | 2018-01-22 15:05:00,452 INFO  [ckan.lib.base]  / render time 0.810 seconds
ckan          | 2018-01-22 15:05:05,411 INFO  [ckan.lib.base]  /organization render time 0.183 seconds
ckan          | 2018-01-22 15:05:07,481 INFO  [ckan.lib.base]  /group render time 0.051 seconds
ckan          | 2018-01-22 15:05:08,114 INFO  [ckan.lib.base]  /about render time 0.027 seconds
solr          | 1952083 INFO  (qtp1989972246-17) [   x:ckan] o.a.s.c.S.Request [ckan]  webapp=/solr path=/select params={facet.limit=50&q=*:*&facet.field=organization&facet.field=groups&facet.field=tags&facet.field=res_format&facet.field=license_id&fl=id+validated_data_dict&start=0&sort=score+desc,+metadata_modified+desc&fq=%2Bdataset_type:dataset&fq=%2Bsite_id:"default"&fq=%2Bstate:active&fq=%2Bpermission_labels:("public")&facet.mincount=1&rows=21&facet=true&wt=json} hits=0 status=0 QTime=1
db            | LOG:  unexpected EOF on client connection with an open transaction
ckan exited with code 0
```

**Comment by amercader** [comment]
I'm also finding this, not just in the search page but randomly. It's frustrating because this just tells us that the ckan container crashed for some reason and left the db one hanging. I'm not really sure what's going on but will investigate.

**Comment by amercader** [comment]
@gerbyzation From the [commit](https://github.com/okfn/docker-ckan/commit/efc5f5e937f1c631bc641bc2f4556feb27388f4d) message:

I nailed down the segmentation faults occurring to using the development server (Paster, --reload option didn't affect). Of course I don't know the exact issue with paster but the underlying issue seems to be Alpine 3.6 + some python packages + a too small stack size.

docker-library/python#211

There are several issues across different projects that confirm that upgrading to Alpine 3.7 solves the issue, so that's what we did.

The upgrade to Alpine 3.7 caused the following issues:

* curl was not part of the image by default, added it to the installed packages

* psycopg2 build failed because of:
        psycopg/psycopg2#594

  The fix is to upgrade the psycopg2 version. This is hardcoded in CKAN's requirements.txt so we work around it changing the version when building the image (this should be upgraded in CKAN core)

Also for good measure we don't use binaries when building the requirements because of the issues described here:

ckan/ckan#3893
